services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusControllerITest.kt

```diff
@@ -194,6 +201,82 @@ class HakemusControllerITest(@Autowired override val mockMvc: MockMvc) : Control
                 hakemusService.getWithPaatokset(id)
             }
         }
+
+        @Test
+        fun `returns ilmoitukset with an excavation notification`() {
+            val hakemus =
+                HakemusFactory.create(
+                    applicationType = ApplicationType.EXCAVATION_NOTIFICATION,
+                    ilmoitukset =
+                        listOf(
+                            IlmoitusFactory.create(type = IlmoitusType.TYO_VALMIS),
+                            IlmoitusFactory.create(dateReported = LocalDate.parse("2024-12-24"))))
+            every { authorizer.authorizeHakemusId(id, PermissionCode.VIEW.name) } returns true
+            every { hakemusService.getWithPaatokset(id) } returns
+                HakemusWithPaatokset(hakemus, listOf())
+
+            val response: HakemusResponse = get(url).andExpect(status().isOk).andReturnBody()
+
+            assertThat(response.ilmoitukset).isNotNull().all {
+                key(IlmoitusType.TOIMINNALLINEN_KUNTO).single().all {
+                    prop(IlmoitusResponse::dateReported).isEqualTo(LocalDate.parse("2024-12-24"))
+                    prop(IlmoitusResponse::reportedAt)
+                        .isEqualTo(IlmoitusFactory.DEFAULT_REPORT_TIME)
+                }
+                key(IlmoitusType.TYO_VALMIS).single().all {
+                    prop(IlmoitusResponse::dateReported).isEqualTo(IlmoitusFactory.DEFAULT_DATE)
+                    prop(IlmoitusResponse::reportedAt)
+                        .isEqualTo(IlmoitusFactory.DEFAULT_REPORT_TIME)
+                }
+            }
+            verifySequence {
+                authorizer.authorizeHakemusId(id, PermissionCode.VIEW.name)
+                hakemusService.getWithPaatokset(id)
+            }
+        }
+
+        @Test
+        fun `returns empty map in ilmoitukset when kaivuilmoitus has none`() {
```

Comment:
There are now both "excavation notification" and "kaivuilmoitus" in test names. Maybe we should decide which one to use?

Reply:
I favor using finnish terms for the domain terminology. For several reasons, but mostly because they're the best defined.

I changed the test names in this PR to say "kaivuilmoitus" and "johtoselvityshakemus".